### PR TITLE
Adding `buildkite-test-collector`

### PR DESCRIPTION
**Pull request by bstadlbauer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bstadlbauer** [comment]
@conda-forge/help-python, ready for review!

**Comment by bstadlbauer** [comment]
@conda-forge/staged-recipes, ready to review

**Comment by bstadlbauer** [comment]
@conda-forge/help-python, ready to review :-)

File: recipes/buildkite-test-collector/LICENSE [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Copyright 2022 Buildkite
```
Comment (ocefpaf) [inline recipes/buildkite-test-collector/LICENSE:1]: You can delete this file. The license is in the tarball.

File: recipes/buildkite-test-collector/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/buildkite-test-collector-{{ version }}.tar.gz
+  sha256: 555b5b7ef771e4ced23e004899c18f837faaa3b6167f3d129b871f8c2419abdb
+
+build:
+  skip: True  # [win]
```
[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/buildkite-test-collector/meta.yaml:13]: No need to skip. Make it noarch and, while the build will fail the final noarch package will be installable on Windows.
```suggestion
  noarch: python
```

File: recipes/buildkite-test-collector/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python
```
Comment (ocefpaf) [inline recipes/buildkite-test-collector/meta.yaml:19]:
```suggestion
    - python >=3.8
```

File: recipes/buildkite-test-collector/LICENSE (hunk 1 above)
Comment (bstadlbauer) [reply recipes/buildkite-test-collector/LICENSE]: Thank you, deleted it 👍

File: recipes/buildkite-test-collector/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bstadlbauer) [reply recipes/buildkite-test-collector/meta.yaml]: Oh great, thank you! Done 👍

File: recipes/buildkite-test-collector/meta.yaml (hunk 3 above)
Comment (bstadlbauer) [reply recipes/buildkite-test-collector/meta.yaml]: Done

File: recipes/buildkite-test-collector/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  skip: True  # [win]
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python 
+    - pip
+  run:
+    - python
```
[the remark this answers is not in the document]
Comment (bstadlbauer) [reply recipes/buildkite-test-collector/meta.yaml]: Done

**Comment by bstadlbauer** [comment]
@ocefpaf Thank you for the quick review, addressed your feedback

**Pull request merged**